tests/client/amo/components/TestLandingPage.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import {
+  renderIntoDocument,
+  findRenderedComponentWithType,
+} from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import { Provider } from 'react-redux';
+
+import * as landingActions from 'amo/actions/landing';
+import {
+  LandingPageBase,
+  mapStateToProps,
+  singularizeAddonType,
+} from 'amo/components/LandingPage';
+import createStore from 'amo/store';
+import I18nProvider from 'core/i18n/Provider';
+import { fakeAddon } from 'tests/client/amo/helpers';
+import { getFakeI18nInst } from 'tests/client/helpers';
+
+
+describe('<LandingPage />', () => {
+  function render({ ...props }) {
+    const initialState = { api: { clientApp: 'android', lang: 'en-GB' } };
+
+    return findDOMNode(findRenderedComponentWithType(renderIntoDocument(
+      <Provider store={createStore(initialState)}>
+        <I18nProvider i18n={getFakeI18nInst()}>
+          <LandingPageBase i18n={getFakeI18nInst()} {...props} />
+        </I18nProvider>
+      </Provider>
+    ), LandingPageBase));
+  }
+
+  it('renders a LandingPage with no addons set', () => {
+    const root = render({
+      addonType: 'extension',
+    });
+
+    assert.include(root.textContent, 'Featured extensions');
+    assert.include(root.textContent, 'More featured extensions');
+  });
+
+  it('renders a LandingPage with themes HTML', () => {
+    const root = render({
+      addonType: 'theme',
+    });
+
+    assert.include(root.textContent, 'Featured themes');
+    assert.include(root.textContent, 'More featured themes');
+  });
+
+  it('renders each add-on when set', () => {
+    const props = {
```

Comment:
I feel like this would be a great place to use `store.dispatch()` to populate the data because it will make sure that the rendering is tied directly to realistic state data. You can still still use `...fakeAddon` of course.